File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.cs
using System;

public class Solution {
    public double FindMedianSortedArrays(int[] nums1, int[] nums2) {
        int m = nums1.Length, n = nums2.Length;
        int[] merged = new int[m + n];
        int i = 0, j = 0, k = 0;

        while (i < m && j < n) {
            if (nums1[i] < nums2[j]) {
                merged[k++] = nums1[i++];
            } else {
                merged[k++] = nums2[j++];
            }
        }
        while (i < m) merged[k++] = nums1[i++];
        while (j < n) merged[k++] = nums2[j++];

        int totalSize = merged.Length;
        if (totalSize % 2 == 0) {
            return (merged[totalSize / 2 - 1] + merged[totalSize / 2]) / 2.0;
        } else {
            return merged[totalSize / 2];
        }
    }
}

File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.rs
impl Solution {
    pub fn find_median_sorted_arrays(nums1: Vec<i32>, nums2: Vec<i32>) -> f64 {
        let mut merged = Vec::new();
        let (mut i, mut j) = (0, 0);

        while i < nums1.len() && j < nums2.len() {
            if nums1[i] < nums2[j] {
                merged.push(nums1[i]);
                i += 1;
            } else {
                merged.push(nums2[j]);
                j += 1;
            }
        }
        merged.extend_from_slice(&nums1[i..]);
        merged.extend_from_slice(&nums2[j..]);

        let total_size = merged.len();
        if total_size % 2 == 0 {
            (merged[total_size / 2 - 1] as f64 + merged[total_size / 2] as f64) / 2.0
        } else {
            merged[total_size / 2] as f64
        }
    }
}
